website/src/arcade/eventEmitters.ts

```diff
@@ -0,0 +1,30 @@
+class SimpleEventEmitter<S> {
+  cbs: ((x: S) => void)[] = [];
+
+  subscribe(cb: (x: S) => void) {
+    this.cbs.push(cb);
+
+    return {
+      unsubscribe: () => {
+        const indexToRemove = this.cbs?.indexOf(cb) ?? -1;
+        if (indexToRemove >= 0) this.cbs?.splice(indexToRemove, 1);
```

Comment:
lol I copy-pasted this from another codebase of mine. `cbs` is instantiated with an empty array, so that should be good to go in the first place – I actually think we can just drop the optional chaining, and be confident `cbs` is always existent.

Reply:
I'll hit that in a follow up PR.